python/pyspark/mllib/tests.py

```diff
@@ -1041,10 +1054,12 @@ def test_trainOn_model(self):
         self.ssc.start()
 
         # Give enough time to train the model.
-        self._ssc_wait(t, 6.0, 0.01)
-        finalModel = stkm.latestModel()
-        self.assertTrue(all(finalModel.centers == array(initCenters)))
-        self.assertEquals(finalModel.clusterWeights, [5.0, 5.0, 5.0, 5.0])
+        def termCheck():
+            finalModel = stkm.latestModel()
+            all(finalModel.centers == array(initCenters)) and \
+            finalModel.clusterWeight == [5.0, 5.0, 5.0, 5.0]
+        self._ssc_wait_checked(t, 20.0, termCheck)
+        self.assertTrue(termCheck())
```

Comment:
For these tests, they should pass whenever all batches have been processed, so the current setup should be safe.  I'm actually thinking of copying the checks so that assertions print out more useful error messages.  (I don't see a great way to avoid copying the checks if I want them for both early stopping & useful error messages.)